hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4120,10 +4123,35 @@ func reconcileOpenshiftMonitoringNetworkPolicy(policy *networkingv1.NetworkPolic
 	return nil
 }
 
-func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyperv1.HostedCluster) error {
+func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyperv1.HostedCluster, managementClusterNetwork *configv1.Network) error {
 	protocolTCP := corev1.ProtocolTCP
 	protocolUDP := corev1.ProtocolUDP
 	protocolSCTP := corev1.ProtocolSCTP
+
+	blockedNetworks := []string{}
+	if len(managementClusterNetwork.Spec.ClusterNetwork) > 0 {
+		for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+			blockedNetworks = append(blockedNetworks, network.CIDR)
+		}
+	}
+	if len(managementClusterNetwork.Spec.ServiceNetwork) > 0 {
+		for _, network := range managementClusterNetwork.Spec.ServiceNetwork {
+			blockedNetworks = append(blockedNetworks, network)
+		}
+	}
+
+	if len(managementClusterNetwork.Spec.ServiceNetwork) > 0 {
+		clusterNetworks := []string{}
+		for _, network := range managementClusterNetwork.Spec.ClusterNetwork {
+			clusterNetworks = append(clusterNetworks, network.CIDR)
+		}
+	}
+	policy.Spec.PolicyTypes = []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress}
+	policy.Spec.PodSelector = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			hyperv1.InfraIDLabel: hcluster.Spec.InfraID,
+		},
```

Comment:
There might be other pods outside of the KubeVirt vms that match this selector within the hcp namespace.

Is there a way to ensure we target just the VMs of this specific cluster?

Reply:
currently only virt-launcher pods are using that label, but to be on the safe side, i've added the `"kubevirt.io": "virt-launcher"` label to the podSelector.